base-template.sh

```diff
@@ -421,19 +439,52 @@ update_shared_hooks_if_appropriate() {
 #   1 in case a hook fails, 0 otherwise
 #####################################################
 execute_shared_hooks() {
-    for SHARED_ROOT in ~/.githooks/shared/*; do
-        REMOTE_URL=$(cd "$SHARED_ROOT" && git config --get remote.origin.url)
+    # split on comma and newline
+    IFS=",
+    "
+    for SHARED_REPO in $SHARED_REPOS_LIST; do
+
+        set_shared_root "$SHARED_REPO"
+
+        # Fail if the shared root is not available (if enabled)
+        FAIL_ON_NOT_EXISTING=$(git config --global --get githooks.failOnNotExistingSharedHooks)
+        
+        if [ "$FAIL_ON_NOT_EXISTING" = "true"]; then
+            if [ ! -d "$SHARED_ROOT" ] || [ ! -d "$SHARED_ROOT/.git" ]; then
+                echo "! Failed to execute shared hook: $SHARED_REPO"
+                echo "! Its is not available. Do run:"
+                echo "!   \$ git hooks shared update"
+                return 1
+            fi
+        fi
+
+        # Note: GIT_DIR might be set (?bug?) (actually the case for post-checkout hook)
+        # which means we really need a `-f` to sepcify the actual config!
+        REMOTE_URL=$(cd "$SHARED_ROOT" && git config -f "$SHARED_ROOT/.git/config" --get remote.origin.url)
         ACTIVE_REPO=$(echo "$SHARED_REPOS_LIST" | grep -o "$REMOTE_URL")
         if [ "$ACTIVE_REPO" != "$REMOTE_URL" ]; then
```

Comment:
This here is meant to filter out repositories that are in the shared hooks root folder but we don't have them on the `$SHARED_REPOS_LIST` list anymore. With your change, is this still needed?

Reply:
not really, but if somebody messes around with the repo in the shared root folder, this asserts that the url in the root is consistent with what we want from the `.shared`. I would leave it in... ?